packages/react-next/src/Utilities/grid/Grid.base.tsx

```diff
@@ -1,98 +1,76 @@
 import * as React from 'react';
-import {
-  getId,
-  toMatrix,
-  classNamesFunction,
-  getNativeProps,
-  htmlElementProperties,
-  initializeComponentRef,
-} from '../../Utilities';
+import { toMatrix, classNamesFunction, getNativeProps, htmlElementProperties } from '../../Utilities';
 import { FocusZone } from '../../FocusZone';
-import { IGrid, IGridProps, IGridStyleProps, IGridStyles } from './Grid.types';
-
+import { IGridProps, IGridStyleProps, IGridStyles } from './Grid.types';
+import { useId } from '@uifabric/react-hooks';
 const getClassNames = classNamesFunction<IGridStyleProps, IGridStyles>();
 
-export class GridBase extends React.Component<IGridProps, {}> implements IGrid {
-  private _id: string;
-
-  constructor(props: IGridProps) {
-    super(props);
-
-    initializeComponentRef(this);
-    this._id = props.id || getId();
-  }
-
-  public render(): JSX.Element {
-    const props = this.props;
-    const {
-      items,
-      columnCount,
-      onRenderItem,
-
-      // tslint:disable:deprecation
-      ariaPosInSet = props.positionInSet,
-      ariaSetSize = props.setSize,
-      // tslint:enable:deprecation
+export const GridBase = React.forwardRef((props: IGridProps, ref: React.Ref<HTMLTableElement>) => {
+  const id = props.id || useId();
```

Comment:
You have to unconditionally generate the default ID first; otherwise you could get a conditional hook call error if `props.id` changes.
```suggestion
  const defaultId = useId();
  const id = props.id || defaultId;
```